bin/findbadmath.scala

```diff
@@ -42,15 +44,15 @@ for {
 
 // now do the StrictMath check
 
-for{path <- Process("find netlogo-core/src netlogo-gui/src -name *.java").lineStream
+for{path <- Process("find netlogo-core/src netlogo-gui/src -name *.java").lazyLines
     if !path.containsSlice("/gl/render/")  // we don't care if OpenGL stuff is strictfp
+    if !path.containsSlice("Activation.java") // Activation uses Math.max, on "safe" data
```

Comment:
I'm just curious - why did this file exclusion need to be moved?

Reply:
Activation.java is in the set of paths, and should be skipped.
It was in the   if(io.Source.fromFile(path).getLines which is searching through the lines in a file.
The string “Activation.java” will not necessarily be in the file Activation.java.
 
for{path <- Process("find netlogo-core/src netlogo-gui/src -name *.java").lazyLines
    if !path.containsSlice("/gl/render/")  // we don't care if OpenGL stuff is strictfp
    if !path.containsSlice("Activation.java") // Activation uses Math.max, on "safe" data
    if path != "src/org/nlogo/headless/TestCommands.java"}
  // this isn't the absolutely correct check to be doing, but it seems like a good enough heuristic
  // for now - ST 5/8/03
  if(io.Source.fromFile(path).getLines
     .filter(!_.containsSlice("Mathematica"))
     .filter(!_.containsSlice("DummyMath"))
     .exists(_.matches(""".*[^t]Math.*""")))
    println("needs StrictMath: " + path)